From: Jason Rhinelander <jason@imaginary.ca>
Date: Wed, 11 Dec 2024 18:21:37 -0400
Subject: stable branch library compatibility fixes

Patches stable to make it compatible with the latest versions of
dependent libraries.

- bump oxen-encoding to latest
- bump oxen-logging to latest
- bump to C++20
- patch fix everything that broke
---
 llarp/quic/connection.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/llarp/quic/connection.cpp b/llarp/quic/connection.cpp
index f9567f4..02393dd 100644
--- a/llarp/quic/connection.cpp
+++ b/llarp/quic/connection.cpp
@@ -90,7 +90,7 @@ namespace llarp::quic
         void* user_data)
     {
       std::basic_string_view data{rawdata, rawdatalen};
-      LogTrace("Receiving crypto data @ level ", crypto_level, " ", buffer_printer{data});
+      LogTrace("Receiving crypto data @ level ", static_cast<int>(crypto_level), " ", buffer_printer{data});
 
       auto& conn = *static_cast<Connection*>(user_data);
       switch (crypto_level)
@@ -150,7 +150,7 @@ namespace llarp::quic
           break;
 
         default:
-          LogWarn("Unhandled crypto_level ", crypto_level);
+          LogWarn("Unhandled crypto_level ", static_cast<int>(crypto_level));
           return FAIL;
       }
       conn.io_ready();
